charts/tenant-base/chart/templates/deployment.yaml

```
@@ -14,10 +14,15 @@ spec:
       {{- include "chart.selectorLabels" (dict "name" $deploymentName) | nindent 6 }}
   template:
     metadata:
-      {{- with .podAnnotations }}
       annotations:
-        {{- toYaml . | nindent 8 }}
-      {{- end }}
+        {{- /* To ensure pods are recreated when a configmap changes that the pod mounts, we annotate the pods with the checksums of the configmaps. */}}
+          {{- range $_, $allConfigMap := $.Values.configMaps -}}
+            {{- range $_, $deploymentConfigMap := $val.configMaps -}}
+              {{- if eq $allConfigMap.name $deploymentConfigMap.name }}
+                {{- printf "%s/%s: %s" "checksum/configmap" $allConfigMap.name ($allConfigMap | toYaml | toString | sha256sum) | nindent 8 }}
```

I am sorry, my suggestion was slightly off, we do not need the `configmap` that should represent the name of the configmap.. sorry :/
```suggestion
                {{- printf "%s/%s: %s" "checksum" $allConfigMap.name ($allConfigMap | toYaml | toString | sha256sum) | nindent 8 }}
```

---

Well, I still think that the "configmap" string should remain so that we know the type of object that created the checksum. If we at some point do the same with secrets it would be nice to be able to tell the difference.
In general i think more descriptive naming is always prefered.

---

That is a fair point (y)